homeassistant/components/custom_card.py

```diff
@@ -0,0 +1,105 @@
+"""
+Show custom full-cards (ha-cards) and sate-cards on Home Assistant frontend.
+
+For more details about this component, please refer to the documentation
+at https://home-assistant.io/components/custom_card/
+"""
+import asyncio
+import json
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import http
+from homeassistant.const import HTTP_BAD_REQUEST
+import homeassistant.helpers.config_validation as cv
+
+DOMAIN = 'custom_card'
+DEPENDENCIES = ['http']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_FULL_CARD = 'full_card'
+CONF_STATE_CARD = 'state_card'
+CONF_MORE_INFO_CARD = 'more_info_card'
+CONF_CONFIG = 'config'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        cv.slug: vol.Any({
+            vol.Optional(CONF_FULL_CARD): cv.string,
+            vol.Optional(CONF_STATE_CARD): cv.string,
+            vol.Optional(CONF_MORE_INFO_CARD): cv.string,
+            vol.Optional(CONF_CONFIG): cv.match_all,
+        }, None)
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Initialize custom card."""
+    card_configs = {}
+
+    for object_id, cfg in config[DOMAIN].items():
+        if not cfg:
+            cfg = {}
+
+        full_card = cfg.get(CONF_FULL_CARD)
+        state_card = cfg.get(CONF_STATE_CARD)
+        more_info_card = cfg.get(CONF_MORE_INFO_CARD)
+        config = cfg.get(CONF_CONFIG)
+
+        if full_card is None and state_card is None:
+            _LOGGER.error("Entity config must contain full_card " +
+                          "and/or state_card (%s)", object_id)
+            return False
+
+        if full_card:
+            entity_id = 'custom_full_card.{}'.format(object_id)
+            state = full_card
+        else:
+            entity_id = 'custom_state_card.{}'.format(object_id)
+            state = state_card
+
+        attributes = {}
+        if full_card:
+            attributes['custom_ui_full_card'] = full_card
+        if state_card:
+            attributes['custom_ui_state_card'] = state_card
+        if more_info_card:
+            attributes['custom_ui_more_info_card'] = more_info_card
+
+        if config:
+            card_configs[entity_id] = json.dumps(config)
+
+        hass.states.async_set(entity_id, state, attributes)
```

Comment:
We will need 2 components. We can't have 1 component create entities that are under another domain.